### PR TITLE
MouseInput for Loading Screen

### PR DESCRIPTION
**Pull request by TheMerkyShadow**
This will allow server owners to design loading screen with the ability to detect the X and Y and also when a users presses a Mouse Left+Right. This can be very useful and would make loading screen less static.

Only issue is sometimes it doesn't get called. But I think it has to do with the high intensive loading.

### CODE REVIEW

**Comment by neico** [comment]
I'd rename it to "MouseInput for LoadingScreen" then.
pretty much the same goes for the commit messages (which you should squash as well)

And the loading screen isn't always responsive as source1 runs pretty much everything on the same cpu core, so the loading screen becomes unresponsive if there's a lot of things to do, and changing mounts, grabbing snapshots or fetching the latest workshop data can be quite resource intensive.

**Comment by TheMerkyShadow** [comment]
@neico Thanks For The Suggestion.
Also I expected as much.

**Comment by TheMerkyShadow** [comment]
I really don't understand why this is being ignored. Its a fairly simple mod and its not much.

**Comment by Kefta** [comment]
No pull requests are being merged

File: garrysmod/lua/menu/loading.lua [hunk 1]
```diff
@@ -241,15 +241,15 @@ function UpdateLoadPanel( strJavascript )
...
+	pnlLoading.JavascriptRun = string.format( "if ( window.MouseInput ) MouseInput( %i, %i, \"%s\", \"%s\" );", gui.MouseX(), gui.MouseY(), input.IsMouseDown(MOUSE_LEFT), input.IsMouseDown(MOUSE_RIGHT) )
 end
 
 function GameDetails( servername, serverurl, mapname, maxplayers, steamid, gamemode )
 
 	if ( engine.IsPlayingDemo() ) then return end
 
 	g_ServerName	= servername
-	g_MapName		= mapname
+	g_MapNamew		= mapname
```
[the remark this answers is not in the document]
Comment (TheMerkyShadow) [reply garrysmod/lua/menu/loading.lua]: nah I messed that up check my most recent commit.

**Comment by TheMerkyShadow** [comment]
@Kefta I think you accepted an old commit in my pull request. Check my most latest change 6a33984
And Thank You.

**Comment by Kefta** [comment]
Yeah I didn't mean to leave a review before, my bad.

**Comment by TheMerkyShadow** [comment]
Its pretty sad @robotboy655  and @Kefta that when somebody adds something useful and works it doesn't get added while useless fixes in TTT get updated immediately.

**Comment by Kefta** [comment]
Bad king handles TTT requests - it's pretty much separate from this repo.

**Comment by Tenrys** [comment]
This seems really hacky, both as an implementation and something to implement on web pages.
Besides, loading screens are locked to the main thread so it's still inconvenient.

**Comment by TheMerkyShadow** [comment]
@Tenrys Seems like a narrow minded person like yourself doesn't have the ability to understand that this implementation gives loading screen designers thousands of new abilities that were once thought to be impossible. Test the code yourself and its "hackyness" before commenting on a random pull request you know absolutely nothing about.

**Comment by Tenrys** [comment]
Was the personal attack really necessary? I'm not contesting the fact that being able to send input to your loading screen is a good idea and nice addition, but it should be done at an engine level and not in a way that seems like a heavy workaround: you are merely executing a custom, specifically named function every frame when actual mouse input events could be used as it should be if this feature were to be implemented properly.

**Comment by TheMerkyShadow** [comment]
It was actually necessary to defend it because the performance loss is nil. And not even something you should even be mentioning. And felt it was a stupid statement.

**Comment by TheMerkyShadow** [comment]
1). Obviously
2). var = document.elementFromPoint(x, y)
3). And to the IsMouseDown I felt that was better than an number.

Sorry I'm salty just upset this is sitting here for a very long time. When in reality this should have been added a long time ago.

**Comment by Mijyuoon** [comment]
> 2). var = document.elementFromPoint(x, y)

Does outdated as hell Awesomium even support that?

> 3). And to the IsMouseDown I felt that was better than an number.

… what? Why not pass it as, you know, *a boolean* like it already is?

**Comment by TheMerkyShadow** [comment]
1) Yes it does.
2) Your probably right

**Comment by TheMerkyShadow** [comment]
I've tested it a long time ago with an if statement and it seemed to work fine.

**Comment by Mijyuoon** [comment]
Wait… I just realized that this actually breaks shit badly:
```lua
function PANEL:MouseInput()
	pnlLoading.JavascriptRun = string.format( "if ( window.MouseInput ) MouseInput( %i, %i, \"%s\", \"%s\" );", gui.MouseX(), gui.MouseY(), input.IsMouseDown(MOUSE_LEFT), input.IsMouseDown(MOUSE_RIGHT) )
end
```

It will prevent function `GameDetails` defined below from working because it sets same variable, which will get overwritten every tick with this. What you should've done is the following:
```lua
function PANEL:MouseInput()
	self:RunJavascript( string.format( "if ( window.MouseInput ) MouseInput( %i, %i, \"%s\", \"%s\" );", gui.MouseX(), gui.MouseY(), input.IsMouseDown(MOUSE_LEFT), input.IsMouseDown(MOUSE_RIGHT) ) )
end
```

**Comment by TheMerkyShadow** [comment]
Woah I would have never seen that

**Comment by robotboy655** [comment]
I will end this pointless debate with this:
It's not going to be merged.

If it was planned to allow mouse input on loading screens, it would've been done when HTML loading screens were added.

**Pull request closed without merging**